### PR TITLE
Enable new CI pipeline with github actions

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

File: .github/workflows/reviewdog.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: reviewdog
+on: [pull_request]
```
Comment (boehmseb) [inline .github/workflows/reviewdog.yml]: Is there a difference between this syntax and the one used in `pre-commit.yml`?

**Review by boehmseb — APPROVED** [review]
Looks good to me. I'm just confused why the _pylint_ and _mypy_ reports are listed under _Python CI_ when they are actually executed by _reviewdog_. Why is this?

File: .github/workflows/reviewdog.yml (hunk 1 above)
Comment (vulder) [reply .github/workflows/reviewdog.yml]: somehow they are always shown under the first action tab, I guess reviewdog posting action results somehow is responsible for this. It's the same for benchbuilds gh actions.

Comment (vulder) [reply .github/workflows/reviewdog.yml]: I tried different setup/naming schemas but the runs get always attributed to the first action... no idea why

**Pull request merged**